test/lib/utils/swapHelpers.ts

```diff
@@ -19,15 +19,46 @@ import {
 } from '../../../src';
 import { sendTransactionGetBalances } from '../../lib/utils/helper';
 
-export async function assertSwapExactIn(
-    contractToCall: Address,
-    client: Client & PublicActions & TestActions & WalletActions,
-    rpcUrl: string,
-    chainId: ChainId,
-    swap: Swap,
-    wethIsEth: boolean,
+// Helper function to check if two BigInts are within a given percentage
+function areBigIntsWithinPercent(
+    value1: bigint,
+    value2: bigint,
+    percent: number,
+): boolean {
+    if (percent < 0) {
+        throw new Error('Percent must be non-negative');
+    }
+    const difference = value1 > value2 ? value1 - value2 : value2 - value1;
+    const percentFactor = BigInt(Math.floor(percent * 1e8));
+    const tolerance = (value2 * percentFactor) / BigInt(1e10);
+    return difference <= tolerance;
+}
+
+export async function assertSwapExactIn({
+    contractToCall,
+    client,
+    rpcUrl,
+    chainId,
+    swap,
+    wethIsEth,
     usePermit2Signatures = false,
-) {
+    outputTest = {
+        isExactOut: true,
+        percentage: 0,
+    },
```

Comment:
I get the idea of differentiating exact match and non-exact match assertions, but I think `isExactOut` flag in an `assertSwapExactIn` check is a bit odd 😅 (same goes for `assertSwapExactOut` with `isExactIn`).
I'd go with simply an optional `percentage`, `withinPercentage` or even `tolerance`. When defined, it specifies the acceptable tolerance and when undefined performs an exact match assertion.
What do you think?

Reply:
Ahh, yeah, I didn't think of it that way! I think its still useful to make it more explicit about where the percentage is changed so I've kept a flag but changed the naming.